configData.py

```diff
@@ -25,6 +25,13 @@
 # Default is 300s.
 minVideoLength: int = 300
 
+# Because of how BERTopic works, there needs to be a minimum number of sentences per window of time.
+# When attempting to automatically segment the transcript into sentences using spaCy,
+# If there are sentences longer than this duration, the transcript data might not be suitable for this process.
+# Instead, the transcript will be segmented based on the `WINDOW_SIZE`.
+# Defaults to 120s, must be >=60.
+maxSentenceDuration = 120
```

Comment:
So the way I understood is Sentence segmentation for a single caption file could be segmented using `spaCy` or hardcoded 30sec window segmentation?

Reply:
If that is true then we want to track that, so I feel having a log statement if segmentation done purely via `spaCy` and/or via 30sec chunk should be logged. I don't what to complicate things here add what is simplest here.

Reply:
Yes. It depends on the quality of the captions. If sentence segmentation is successful, then spaCy is used. If not, the fallback is to using the hardcoded window. This way some segmentation is done instead of just failing outright.

Reply:
The function `validateProcessedSentences` logs whether the segmentation approach was successful or not. 
The commit I just made should be explicit about stating that spaCy was the one used for segmentation when it worked. 

It is one or the other, if spaCy failed to segment and any sentence is found to be over two minutes, the whole transcript is hardcoded to use 30s segments nstead. It is possible to have a sort of hybrid mix of spaCy and hardcoded segmentation, with an added increase in complexity to handle that.

Reply:
Ok, That's what I was thinking whether it segmentation is some kind of Hybrid or it's either that or this. I feel that is ok for now. When we do some testing and based on the feedback received from Q Generation we can comeback to it.